bin/identity.sh

```diff
@@ -0,0 +1,424 @@
+   
+        exit;
+    }
+  
+    $username = get_option_value($options, 'username', '', false, true, "Enter the username e.g. -u user@example.com");
+
+    $host = rcube_utils::get_host($options);
+    $user = get_user($username, $host);
+
+    if (!$user) {
+      exit;
+    }
+
+    $identities = $user->list_identities(null, true);
+
+    echo_identities($identities);
+}
+
+function delete_identity($options) {
+    if (count($options) === 1) {
+        echo "delete\n";
+        echo "======\n\n";
+        echo "    Deletes an identity.\n";
+        echo "    This sets 'del' value of an identity to 1 making it inactive, but it is not removed from the database. \n\n\n";
+        echo "    Mandatory arguments:\n\n";
+        echo "    -u\n";
+        echo "        username - the identity holder e.g. -u mainmail@example.com\n\n";
+        echo "    -i\n";
+        echo "        identity id - the id of the identity being queried e.g. -i 70) use list sub-command to get identity id\n\n";
+  
+        exit;
+    }  
+
+    $username = get_option_value($options, 'username', '', false, true, "Enter the username e.g. -u user@example.com");
+    $identity_id = get_option_value($options, 'identity_id', '', false, true, "Enter the identity id e.g. -i 70");
+
+    $host = rcube_utils::get_host($options);
+    $user = get_user($username, $host);
+
+    if (!$user) {
+        exit;
+    }
+
+    $identity = $user->delete_identity($identity_id);
+
+    if (!$identity) {
+        rcube::raise_error("Invalid identity ID.");
+    
+        exit;
+    }
+
+    echo "Identity deleted.\n";
+}
+
+function add_identity($options) {
+    if (count($options) === 1) {
+        echo "add\n";
+        echo "===\n\n";
+        echo "    Creates a new identity for a given user.\n\n\n";
+        echo "    Mandatory arguments:\n\n";
+        echo "    -u\n";
+        echo "        username - the identity holder e.g. -u mainmail@example.com\n\n";
+        echo "    -e\n";
+        echo "        email - the email of the identity e.g. identityemail@example.com\n\n";
+        echo "    -n\n";
+        echo "        the name of the identity - e.g. -n 'John Smith'\n\n\n";
+        echo "    Optional arguments:\n\n";
+        echo "    -o\n";
+        echo "        organization - e.g. -o 'Your Organization Name'\n\n";
+        echo "    -r\n";
+        echo "        reply-to email - e.g. -r replytothisemail@example.com\n\n";
+        echo "    -b\n";
+        echo "        bcc email - e.g. -b bcc@example.com\n\n";
+        echo "    -s\n";
+        echo "        signature - e.g. -s 'Sincerely, John Smith'\n\n";
+        echo "    -H\n";
+        echo "        is signature HTML (empty value or 1 for yes, 0 for no) e.g. -H\n\n";
+        echo "    -S\n";
+        echo "        should this be set as a default identity for the user (only 1 available so it disables all other. Empty value or 1 for yes, 0 for no) e.g. -S 1\n\n";;
```

Comment:
I propose to take this out. There is nothing wrong with it, but the mapping between the argument name and identity attribute is repeated, and there is also a lot of copy-pasted code in the arguments description. I think that it is perfectly fine to create an identity, perhaps emitting the ID of it in a format that can be extracted easily. Then, the actual feeding of the identity with data can be accomplished by the update / set functionality.

Reply:
I created a function to avoid repetition